### PR TITLE
Reactify/create workspace

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

File: app/ts/SchemaUtil/SchemaUtil.ts [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+}
+
+function getDataSourcesById(criteria) {
+  return _(criteria)
+    .flatMap(function (criterion) {
+      return criterion.dataSources;
+    })
+    .keyBy('id')
+    .value();
+}
```
Comment (chinook25) [inline app/ts/SchemaUtil/SchemaUtil.ts:554]: move to more global util; also use it in addSubproblemContext

File: app/ts/Workspaces/CreateWorkspace/CreateWorkspaceUtil/CreateWorkspaceUtil.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+): Record<string, TScenarioPvf> {
+  return _(criteria)
+    .mapValues((criterion) => _.omit(criterion.dataSources[0].pvf, 'range'))
+    .pickBy(isUsablePvf)
+    .value();
+}
+
+function isUsablePvf(pvf: TScenarioPvf): pvf is ILinearPvf {
+  //make ts not care about type of PVF FIXME
+  return !!pvf.direction;
```
Comment (DanielReid) [inline app/ts/Workspaces/CreateWorkspace/CreateWorkspaceUtil/CreateWorkspaceUtil.ts:17]: I think we moved from double negation to `Boolean(pvf.direction)`

File: app/ts/Workspaces/CreateWorkspace/TWorkspaceCreateMethod.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export type TWorkspaceCreateMethod =
```
Comment (DanielReid) [inline app/ts/Workspaces/CreateWorkspace/TWorkspaceCreateMethod.ts:1]: `TWorkspaceCreationMethod`

**Review by DanielReid — COMMENTED** [review]
nice

File: app/ts/Workspaces/CreateWorkspace/CreateWorkspaceDialog/CreateWorkspaceDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+          </Grid>
+        </Grid>
+      </DialogContent>
+      <DialogActions>
+        <Button
+          id="add-workspace-button"
+          color="primary"
+          onClick={handleAddButtonClick}
+          variant="contained"
+          disabled={!_.isEmpty(validationErrors)} // FIXME: disable when uploading and not choosing file
```
Comment (kzalite) [inline app/ts/Workspaces/CreateWorkspace/CreateWorkspaceDialog/CreateWorkspaceDialog.tsx:172]: This still needs to be fixed?

File: app/ts/Workspaces/CreateWorkspace/ValidationUtil/ValidationUtilTest.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+describe('ValidationUtil', () => {
+  describe('validateJsonSchema', () => {
+    it('should', () => {
+      fail();
```
Comment (kzalite) [inline app/ts/Workspaces/CreateWorkspace/ValidationUtil/ValidationUtilTest.ts:4]: tests missing

File: node-backend/premades.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  ],
+  tutorials: [
+    {
+      title: 'Lixisenatide simplified',
+      key: 'lixisenatide-simplified',
+      type: 'tutorial'
+    },
+    {
+      title: 'Zinbryta initial assessment simplified',
+      key: 'zynbritaSimplified',
```
Comment (kzalite) [inline node-backend/premades.ts:41]: not that it matters but lixi key is kebab while these are camel

File: node-backend/util.ts [hunk 7]
```diff
@@ -14,11 +14,7 @@ import logger from './logger';
 export function getUser(request: any) {
   if (request.user) {
     return request.user;
-  } else if (
-    request.session &&
-    request.session.user &&
-    request.session.user.id
-  ) {
+  } else if (request.session?.user?.id) {
```
Comment (kzalite) [inline node-backend/util.ts:17]: love it

File: shared/CreateWorkspaceUtil/CreateWorkspaceUtil.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  criteria: Record<string, IUploadProblemCriterion>
+): Record<string, TScenarioPvf> {
+  return _(criteria)
+    .mapValues((criterion) => _.omit(criterion.dataSources[0].pvf, 'range'))
+    .pickBy(isUsablePvf)
+    .value();
+}
+
+function isUsablePvf(pvf: TScenarioPvf): pvf is ILinearPvf {
+  //make ts not care about type of PVF FIXME
```
Comment (kzalite) [inline shared/CreateWorkspaceUtil/CreateWorkspaceUtil.ts:16]: fixme?

File: app/ts/Workspaces/CreateWorkspace/CreateWorkspaceDialog/CreateWorkspaceDialog.tsx (hunk 4 above)
Comment (kzalite) [reply app/ts/Workspaces/CreateWorkspace/CreateWorkspaceDialog/CreateWorkspaceDialog.tsx:172]: is fixed, just comment left

**Pull request merged**